src/reducers/versions.spec.tsx

```diff
@@ -2470,4 +2524,206 @@ describe(__filename, () => {
+
+        const store = configureStore();
+        store.dispatch(
+          actions.loadVersionInfo({
+            version: { ...fakeVersion, id: versionId },
+          }),
+        );
+
+        const typedPosition = position as RelativePathPosition;
+
+        const { dispatch, thunk } = thunkTester({
+          createThunk: () =>
+            _goToRelativeDiff({
+              _getRelativeDiff,
+              _viewVersionFile,
+              currentAnchor,
+              diff,
+              pathList,
+              position: typedPosition,
+              versionId,
+            }),
+          store,
+        });
+
+        await thunk();
+
+        expect(dispatch).toHaveBeenCalledWith(fakeThunk.thunk);
+        expect(_viewVersionFile).toHaveBeenCalledWith({
+          preserveHash: false,
+          selectedPath: path,
+          showDiff: diffPosition,
+          versionId,
+        });
+      },
+    );
+
+    it('dispatches nothing if no anchor or path is returned', async () => {
+      const _getRelativeDiff = jest
+        .fn()
+        .mockReturnValue({ anchor: null, path: null });
+      const versionId = 123;
+
+      const store = configureStore();
+      store.dispatch(
+        actions.loadVersionInfo({ version: { ...fakeVersion, id: versionId } }),
+      );
+
+      const { dispatch, thunk } = thunkTester({
+        createThunk: () =>
+          _goToRelativeDiff({
+            _getRelativeDiff,
+            versionId,
+          }),
+        store,
+      });
+
+      await thunk();
+
+      expect(dispatch).not.toHaveBeenCalled();
+    });
+
+    it('throws an exception if the version is not loaded', async () => {
+      const versionId = 123;
+      const store = configureStore();
+      store.dispatch(
+        actions.loadVersionInfo({ version: { ...fakeVersion, id: versionId } }),
+      );
+
+      const { thunk } = thunkTester({
+        createThunk: () =>
+          _goToRelativeDiff({
+            versionId: versionId + 1,
+          }),
+        store,
+      });
+
+      // TODO: This doesn't work. Maybe we cannot test for an exception in a thunk?
+      expect(async () => {
+        await thunk();
+      }).toThrow('Cannot go to relative diff without a version loaded');
+    });
+
+    it('does not call _getRelativeDiff if the version is not loaded', async () => {
+      const _getRelativeDiff = jest.fn();
+      const versionId = 123;
+
+      const store = configureStore();
+      store.dispatch(
+        actions.loadVersionInfo({ version: { ...fakeVersion, id: versionId } }),
+      );
+
+      const { thunk } = thunkTester({
+        createThunk: () =>
+          _goToRelativeDiff({
+            _getRelativeDiff,
+            versionId: versionId + 1,
+          }),
+        store,
+      });
+
+      // TODO: Without this, the test stops because of the exception,
+      // but this doesn't seem like a good thing to do in a test.
+      try {
+        await thunk();
```

Comment:
I guess you could use `await expect(thunk()).rejects.toThrow()` again but is this test really necessary? A thrown error will stop execution so I don't think you need to worry about rogue dispatches.

Reply:
Good point. I agree that this test isn't really needed.